starlette/requests.py

```diff
@@ -216,9 +215,22 @@ async def stream(self) -> typing.AsyncGenerator[bytes, None]:
 
     async def body(self) -> bytes:
         if not hasattr(self, "_body"):
-            chunks = []
-            async for chunk in self.stream():
-                chunks.append(chunk)
+            if hasattr(self, "_headers"):
+                content_length = self.headers.get("Content-Length")
+                if content_length:
+                    # Allocating a bytearray with a fixed size,
+                    # that matches "content-length" header.
+                    body_bytes = bytearray(int(content_length))
+                    body_index = 0
+                    async for chunk in self.stream():
+                        chunk_length = len(chunk)
+                        body_bytes[body_index : body_index + chunk_length] = chunk
+                        body_index += chunk_length
```

Comment:
Nit (only change if you'd like to): we can do one less addition here.
```suggestion
                        chunk_end_index = body_index + len(chunk)
                        body_bytes[body_index:chunk_end_index] = chunk
                        body_index = chunk_end_index
```